src/dataframe/dataframe.jl

```diff
@@ -6,6 +6,16 @@ An AbstractDataFrame that stores a set of named columns
 The columns are normally AbstractVectors stored in memory,
 particularly a Vector or CategoricalVector.
 
+# Examples
```

Comment:
We have examples below. I think we should not duplicate it. @nalimilan - do you have an opinion?

Reply:
Is the duplication the problem, or is having examples immediately available t..he problem?

<img width="991" alt="Screenshot 2020-05-20 at 10 24 16" src="https://user-images.githubusercontent.com/52289/82429593-15c7d600-9a84-11ea-8f2d-c60e2383f8b0.png">

It's frustrating for a user to ask for help then to have to ask again because the previous answer wasn't helpful.

Reply:
Both I'd say. Examples shouldn't generally be presented in the short help IMHO.

Reply:
I think we'll agree to disagree :) I think almost the opposite!

Reply:
Let's say you've forgotten the syntax for `replace()`. I'd show this:

<img width="977" alt="Screenshot 2020-05-20 at 10 36 11" src="https://user-images.githubusercontent.com/52289/82430816-d13d3a00-9a85-11ea-8e97-23e441812a53.png">

But you would show this:

<img width="977" alt="Screenshot 2020-05-20 at 10 37 15" src="https://user-images.githubusercontent.com/52289/82430909-e31edd00-9a85-11ea-8e89-640759101fc8.png">

I'd say my way was helpful.

Reply:
Actually I think we need clearer guidance regarding what should be put in the extended help in general. Your proposal amounts to putting the full `replace` docstring in the short help. :-)

Reply:
Yes, that's a fair summary! :) My initial understanding of the purpose behind @tholy's "extended help" PR was to provide the user with a readable and useful amount of information in a single screen without forcing them to scroll back through 100s of lines of small print, LaTeX formulas, exhaustive analyses, and scholarly references just to find the basic syntax for the function. I didn't think the concept of "short help without examples" was being proposed...

Reply:
OK. But then you probably should have kept `# Arguments` in the short help in this PR, right? Basically, most docstrings shouldn't be changed at all.

Reply:
"It's not always easy to judge the right amount of help before inserting Extended help, but a rough guide was to make sure you can still see the function definition when the julia> prompt reappears..." - sometimes I may have guessed too quickly that there was too much information for a screen. 

I don't think I'll be doing more of these...